OpenRA.Mods.Common/Activities/FindResources.cs

```diff
@@ -41,21 +41,42 @@ public FindResources(Actor self)
 			domainIndex = self.World.WorldActor.Trait<DomainIndex>();
 		}
 
-		public FindResources(Actor self, CPos avoidCell)
+		public FindResources(Actor self, CPos orderLocation)
 			: this(self)
 		{
-			this.avoidCell = avoidCell;
+			this.orderLocation = orderLocation;
+		}
+
+		protected override void OnFirstRun(Actor self)
+		{
+			// Without this, multiple "Harvest" orders queued directly after each other
+			// will be skipped because the harvester starts off full.
+			if (harv.IsFull)
+				QueueChild(self, new DeliverResources(self));
 		}
 
 		public override Activity Tick(Actor self)
 		{
+			if (ChildActivity != null)
+			{
+				ChildActivity = ActivityUtils.RunActivity(self, ChildActivity);
+				return this;
+			}
+
 			if (IsCanceling)
 				return NextActivity;
 
 			if (harv.IsFull)
 			{
-				// HACK: DeliverResources is ignored if there are queued activities, so discard NextActivity
-				return ActivityUtils.SequenceActivities(self, new DeliverResources(self));
+				return NextActivity;
```

Comment:
I do wonder whether it would be better to queue a child `DeliverResources` here if `NextActivity == null` so that we can keep the same `FindResources` active until the player intervenes.  This would certainly be cleaner if it means we can remove code from `TickIdle`, but i'm not sure whether my suggestion here is sufficient to allow that.

Reply:
...Or possibly even `NextActivity == null || NextActivity is FindResources` and then hopefully get rid of the `OnFirstRun` block above?

Reply:
I went a bit further and removed all the code from `TickIdle` and let `FindResources` handle all harvester automation. This prevents issues with ignored orders like was the case with the old `AttackMove` implementation. Also, after thinking some more about it, I've become convinced that queueing harvester 'runs' is not intuitive and that @matjaeck s suggestion is the more sensible default behaviour, so since I had to refactor anyway I've implemented that as well.

Reply:
I actually quite liked the "runs" idea, but I guess this works too.

Reply:
Ideally, I would like it if you could do both, but I haven't yet come up with a good model for that.

Reply:
Since I keep going back and forth on this I've just made it yaml configurable.

Reply:
TBH I think i'd prefer having just per-cell vs both with a yaml flag (assumiing the default would stay as per-cell) since that keeps the code simpler.

Reply:
I disagree with that. This hardly adds to the code complexity. I actually kind of like how explicit the behaviour is now in the code versus before the refactor. Also, I've made per-run the default again.

Reply:
> Also, I've made per-run the default again.

Defaults to `QueueFullLoad = false` now (which I find great), the previous comments indicate that this shouldn't be a problem.